### PR TITLE
Pure cmake

### PR DESCRIPTION
**Pull request by righetti**
[//]: # "Thanks for your contribution.  To make life of the reviewers easier,"
[//]: # "please give this pull request a meaningful title and provide the"
[//]: # "requested information below."

## Description
The goal is to create an ament-free, pure cmake version of this package that still can be compiled and installed using colcon.
The package can also be compiled independently using cmake only.

[//]: # "Description of what you did.  If it is more complex, consider to add a"
[//]: # "'Summary' section."
* I removed all ament dependencies and the package.xml
* created a pure cmake package following [this cmake tutorial ](https://cmake.org/cmake/help/latest/guide/importing-exporting/index.html#exporting-targets-from-the-build-tree)

## How I Tested
* compiled and installed the package in pure cmake
`mkdir build`
`cd build`
`cmake ..`
`cmake --build .`
`cmake --install . --prefix "/myworspacepath/install" `

* compiled and installed the package using colon and --merge-install
`colcon build --merge-install`

* ran all the tests - made sure I can run the python bindings, etc.

[//]: # "Explain how you tested your changes"

## Do not merge before

[//]: # "Link other open pull request here that need to be merged first."
[//]: # "Remove this section if it does not apply."

- [x] add dependency to mpi_cmake_modules to deal with RPATH
- [x] fix issue with running demos on mac / python3 (`!#/usr/bin/env python3` sanitizes DYLD_LIBRARY_PATH due to SIP and without RPATH properly dealt with libraries are not found
- [x] create macros to find python_install path and move them to mpi_cmake_modules
- [x] test on unbuntu 18.04

## I fulfilled the following requirements

[//]: # "Please make sure you followed these steps before requesting a review."
[//]: # "Check the boxes in the list below, when done."

- [x] All new code is formatted according to our style guide (for C++ run clang-format, for Python, run flake8 and fix all warnings).
- [x] All new functions/classes are documented and existing documentation is updated according to changes.
- [x] No commented code from testing/debugging is kept (unless there is a good reason to keep it).

### CODE REVIEW

**Comment by jviereck** [comment]
I tried to install this on Ubuntu 18.04. When running the `colcon build --merge-install` I end up with problems finding the pybind11 cmake setup:

```bash
$ colcon build --merge-install
Starting >>> solver  
--- stderr: solver
CMake Error at CMakeLists.txt:25 (find_package):
  Could not find a package configuration file provided by "pybind11" with any
  of the following names:

    pybind11Config.cmake
    pybind11-config.cmake
```

I tired to fix this by installing the `pybind11` pip package without success. In addition, I tried to use the pybind11_vendor package. However, this relies on ament again, which was the idea to avoid with the repackaging.

**Comment by jviereck** [comment]
Installing the pybind11 via the apt-get package seems to resolve the last reported issue:

```
sudo apt-get install python-pybind11
```

**Comment by jviereck** [comment]
When building the yaml_utils package on the pure_cmake branch, I was getting 

```bash
$ colcon build --merge-install
Starting >>> yaml_utils
--- stderr: yaml_utils
CMake Error at /usr/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find GTest (missing: GTEST_LIBRARY GTEST_MAIN_LIBRARY)
```

The solution for me was to build the gtest library as outlined here: https://stackoverflow.com/a/52392930

```bash
sudo apt install libgtest-dev build-essential cmake
cd /usr/src/googletest
sudo cmake .
sudo cmake --build . --target install
```

File: momentumopt/colcon.pkg [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "dependencies": ["solver", "solver_lqr", "pybind11", "yaml_utils", "Eigen3"],
+    "build-dependencies": ["mpi_cmake_modules"],
+    "run-dependencies": [],
+    "test-dependencies": ["googletest-distribution"]
+}
```
Comment (righetti) [inline momentumopt/colcon.pkg:6]: is it necessary? I checked on my mac and colcon can build the dependency tree without this. Is it different on Ubuntu?
If I run ```colcon info --packages-select momentumopt``` I get
```
  type: cmake
  name: momentumopt
  dependencies:
    build: Eigen3 GTest mpi_cmake_modules pybind11 solver solver_lqr yaml_utils
    run: Eigen3 GTest mpi_cmake_modules pybind11 solver solver_lqr yaml_utils
```

**Comment by righetti** [comment]
> Installing the pybind11 via the apt-get package seems to resolve the last reported issue:
> 
> ```
> sudo apt-get install python-pybind11
> ```

You would need to do a global install with pip to get the cmake files in the right directory https://github.com/pybind/pybind11/blob/master/docs/installing.rst#include-with-pypi
Not sure what's the best policy for our lab. I used ```brew install pybind11``` on my machine for this reason.

**Comment by MaximilienNaveau** [comment]
@righetti @jviereck @majidkhadiv can you run the demo on solo8?
They should pass.
The one on bolt and solo12 did not finished properly.
The one with bolt hanged (my pc maybe too slow)
The one with solo12 crashed due to some eigen stuff (yaml parsing?)

If you all validated we should merge.

**Comment by jviereck** [comment]
> If you all validated we should merge.

Can you tell me which command to run for this? We fixed some problem with the solo12 demos on the amd-clmc repo. Maybe we have to port it over here as well.

**Comment by MaximilienNaveau** [comment]
```
workspace/src/kino_dynamic_opt/momentumopt/demos
cd 
ipython demo_momentumopt_solo_jump.py
ipython demo_momentumopt_solo_jump_no_lqr.py
```

**Pull request merged**